examples/step-9/step-9.cc

```diff
@@ -734,18 +645,19 @@ namespace Step9
         {
           for (unsigned int j = 0; j < dofs_per_cell; ++j)
             copy_data.cell_matrix(i, j) +=
-              ((advection_directions[q_point] *
+              ((scratch_data.advection_directions[q_point] *
                 scratch_data.fe_values.shape_grad(j, q_point) *
                 (scratch_data.fe_values.shape_value(i, q_point) +
-                 delta * (advection_directions[q_point] *
+                 delta * (scratch_data.advection_directions[q_point] *
                           scratch_data.fe_values.shape_grad(i, q_point)))) *
                scratch_data.fe_values.JxW(q_point));
 
           copy_data.cell_rhs(i) +=
             ((scratch_data.fe_values.shape_value(i, q_point) +
-              delta * (advection_directions[q_point] *
+              delta * (scratch_data.advection_directions[q_point] *
                        scratch_data.fe_values.shape_grad(i, q_point))) *
-             rhs_values[q_point] * scratch_data.fe_values.JxW(q_point));
+             scratch_data.rhs_values[q_point] *
+             scratch_data.fe_values.JxW(q_point));
```

Comment:
I know that the original code wasn't much nicer, but would you mind to beautify this a bit?

Reply:
This is tricky since `scratch_data` makes the lines longer. I'll figure something out!